pinot-core/src/main/java/com/linkedin/pinot/core/query/aggregation/groupby/DictionaryBasedGroupKeyGenerator.java

```diff
@@ -435,26 +403,42 @@ public void remove() {
             }
           }
         } else {
-          if (rawKeys == null) {
-            rawKeys = new int[numValues];
-            for (int j = 0; j < numValues; j++) {
-              int dictId = multiValueDictIds[j];
-              rawKeys[j] = rawKey * cardinality + dictId;
+          int[] multiValueDictIds = _multiValueDictIds[i][index];
```

Comment:
Can we factor this out?

Reply:
It's already factored out as a helper method.